File: typescript/collatz-conjecture/collatz-conjecture.ts
class CollatzConjecture {
  static steps(int: number): number {
    if (int <= 0) {
      throw new Error('Only positive numbers are allowed');
    }
    let steps: number = 0;
    while (int !== 1) {
      int = int % 2 === 0 ? int / 2 : int * 3 + 1;
      steps++;
    }
    return steps;
  }
}

export default CollatzConjecture;

File: go/collatz-conjecture/collatz_conjecture.go
package collatzconjecture

import "fmt"

// CollatzConjecture returns the number of steps for the conjecture bring input to 1.
func CollatzConjecture(input int) (steps int, err error) {
	if input <= 0 {
		return 0, fmt.Errorf("invalid input: %d, must be above 0", input)
	}
	for input != 1 {
		steps++
		if input%2 == 0 {
			input /= 2
		} else {
			input = input*3 + 1
		}
	}
	return steps, nil
}
